gnovm/pkg/gnolang/files_test.go

```diff
@@ -0,0 +1,141 @@
+package gnolang_test
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/gnolang/gno/gnovm/pkg/gnolang"
+	"github.com/gnolang/gno/gnovm/pkg/test"
+	"github.com/stretchr/testify/require"
+)
+
+var withSync = flag.Bool("update-golden-tests", false, "rewrite tests updating Realm: and Output: with new values where changed")
+
+type nopReader struct{}
+
+func (nopReader) Read(p []byte) (int, error) { return 0, io.EOF }
+
+// TestFiles tests all the files in "gnovm/tests/files".
+//
+// Cheatsheet:
+//
+//	fail on the first test:
+//		go test -run TestFiles -failfast
+//	run a specific test:
+//		go test -run TestFiles/addr0b
+//	fix a specific test:
+//		go test -run TestFiles/'^bin1.gno' -short -v -update-golden-tests .
+func TestFiles(t *testing.T) {
+	rootDir, err := filepath.Abs("../../../")
+	require.NoError(t, err)
+
+	opts := &test.TestOptions{
+		RootDir: rootDir,
+		Output:  io.Discard,
+		Error:   io.Discard,
+		Sync:    *withSync,
+	}
+	opts.BaseStore, opts.TestStore = test.Store(
+		rootDir, true,
+		nopReader{}, opts.WriterForStore(), io.Discard,
+	)
+
+	dir := "../../tests/"
+	fsys := os.DirFS(dir)
+	err = fs.WalkDir(fsys, "files", func(path string, de fs.DirEntry, err error) error {
+		switch {
+		case err != nil:
+			return err
+		case path == "files/extern":
+			return fs.SkipDir
+		case de.IsDir():
+			return nil
+		}
+		subTestName := path[len("files/"):]
+		if strings.HasSuffix(path, "_long.gno") && testing.Short() {
+			t.Run(subTestName, func(t *testing.T) {
+				t.Log("skipping in -short")
+			})
```

Comment:
This would make `TestFiles` skip, rather than skipping the individual test. But I think I get what you mean: using t.Skip so that the test is marked as SKIPPED in the output.

Changing t.Log to t.Skip. Thanks!